src/app/modules/create-test-set/create-test-set.component.ts

```diff
@@ -0,0 +1,161 @@
+import { Component, OnInit, group } from '@angular/core';
+import { NgForm } from '@angular/forms';
+import { ImapMailsService } from '../../service/imapemails.service';
+import { MatDialogRef, MatDialog, MatSnackBar } from '@angular/material';
+import { AllTestQuestionComponent } from './../all-test-question/all-test-question.component';
+import _ = require('lodash');
+import { config, limitTime } from '../../config/config';
+import { CommonService } from '../../service/common.service';
+
+@Component({
+    selector: 'app-create-test-set',
+    templateUrl: './create-test-set.component.html',
+    styleUrls: ['./create-test-set.component.scss']
+})
+export class CreateTestSetComponent implements OnInit {
+    testGroup: any[];
+    loading: boolean;
+    limit: any[];
+    testLimit: any;
+    testGroupId: any;
+    testName: any;
+    questions = [];
+    testType: any[];
+    selectedType: any;
+    subjective: boolean;
+    job_profile: any;
+    showmessage = false;
+    message: any;
+    round = []
+    roundId: any;
+    updateData: any;
+    jobProfile: any;
+    private dialogRef: MatDialogRef<any>
+    constructor(private _commonService: CommonService, private dialogRef2: MatDialogRef<any>, private dialog: MatDialog, private _mdSnackBar: MatSnackBar, private apiCall: ImapMailsService) { }
+
+    ngOnInit() {
+        this.loading = true;
+        this.getTestGroup();
+        this.getAllTag();
+        this.limit = limitTime;
+        if (this.updateData) {
+            this.updateData = JSON.parse(JSON.stringify(this.updateData));
```

Comment:
this.updateData = JSON.parse(JSON.stringify(this.updateData)); 
can you explain logic behind this .